fdbserver/SimKmsConnector.actor.cpp

```diff
@@ -71,14 +97,25 @@ struct SimKmsConnectorContext : NonCopyable, ReferenceCounted<SimKmsConnectorCon
 		// Construct encryption keyStore.
 		// Note the keys generated must be the same after restart.
 		for (int i = 1; i <= maxEncryptionKeys; i++) {
+			const int keyLen = SimEncryptKeyCtx::getKeyLen(i);
+			uint8_t key[keyLen];
 			uint8_t digest[AUTH_TOKEN_HMAC_SHA_SIZE];
+
+			ASSERT_EQ(AES_256_KEY_LENGTH, AUTH_TOKEN_HMAC_SHA_SIZE);
 			computeAuthToken({ { reinterpret_cast<const uint8_t*>(&i), sizeof(i) } },
 			                 SHA_KEY,
 			                 AES_256_KEY_LENGTH,
 			                 &digest[0],
 			                 EncryptAuthTokenAlgo::ENCRYPT_HEADER_AUTH_TOKEN_ALGO_HMAC_SHA,
 			                 AUTH_TOKEN_HMAC_SHA_SIZE);
-			simEncryptKeyStore[i] = std::make_unique<SimEncryptKeyCtx>(i, reinterpret_cast<const char*>(&digest[0]));
+			memcpy(&key[0], &digest[0], std::min(keyLen, AUTH_TOKEN_HMAC_SHA_SIZE));
+			// Simulate variable length 'baseCipher' returned by external KMS
```

Comment:
I'm assuming it wouldn't be possible for the key length to be < `AES_256_KEY_LENGTH` right?

Reply:
Just to add more context; the code here is to add **deterministic properties to SimKms**. The way it is achieved is module defines a _stable_ seed value, it is used to generate the encryption key-set which SimKms dole out to its caller. Code uses HMAC_SHA token generation scheme and it uses 32 bytes key-length. 

In summary, the code here is not dealing with external KMS baseCipher keys.

Reply:
Yeah what im saying is in this test the keyLen >= `AES_256_KEY_LENGTH` do we not want to test cases where keyLen < `AES_256_KEY_LENGTH`?

Reply:
The code uses HMAC_SHA token generation that would generate 32 bytes token, so min SimKey baseCipher is 32 bytes at the moment, shrinking would  may cause collisions, we can use some other algo to generate smaller tokens, but for now I think the code is sufficient. 
Would put a TODO to extend it in near futures, if that helps.